cpp/cmake_modules/SetupCxxFlags.cmake

```diff
@@ -270,6 +271,10 @@ if(CXX_SUPPORTS_SSE4_2 AND ARROW_SSE42)
   set(CXX_COMMON_FLAGS "${CXX_COMMON_FLAGS} -msse4.2")
 endif()
 
+if(CXX_SUPPORTS_AVX512 AND ARROW_AVX512)
+  set(CXX_COMMON_FLAGS "${CXX_COMMON_FLAGS} -march=skylake-avx512")
```

Comment:
This is not the same as `-mavx512f` above.

Reply:
skylake-avx512 is a combined options which include basic AVX512 subsets for most AVX512 capable machines. The check_cxx_compiler_flag can only inspect the single define.

echo | gcc -dM -E - -march=skylake-avx512 | grep AVX512
#define __AVX512F__ 1
#define __AVX512BW__ 1
#define __AVX512VL__ 1
#define __AVX512CD__ 1
#define __AVX512DQ__ 1